File: Editorial/73_Set_Matrix_Zeroes/Go/1_Additional_Memory_Approach.go
func setZeroes(matrix [][]int) {
    rows := make(map[int]bool)
    cols := make(map[int]bool)
    // Essentially, we mark the rows and columns that are to be made zero
    for i := 0; i < len(matrix); i++ {
        for j := 0; j < len(matrix[0]); j++ {
            if matrix[i][j] == 0 {
                rows[i] = true
                cols[j] = true
            }
        }
    }
    // Iterate over the array once again and using the rows and cols sets, update the elements
    for i := 0; i < len(matrix); i++ {
        for j := 0; j < len(matrix[0]); j++ {
            if rows[i] || cols[j] {
                matrix[i][j] = 0
            }
        }
    }
}

File: Editorial/0073_Set_Matrix_Zeroes/CSharp/1_Additional_Memory_Approach.cs
public class Solution {
    public void SetZeroes(int[][] matrix) {
        int R = matrix.Length;
        int C = matrix[0].Length;
        HashSet<int> rows = new HashSet<int>();
        HashSet<int> cols = new HashSet<int>();
        // Essentially, we mark the rows and columns that are to be made zero
        for (int i = 0; i < R; i++) {
            for (int j = 0; j < C; j++) {
                if (matrix[i][j] == 0) {
                    rows.Add(i);
                    cols.Add(j);
                }
            }
        }

        // Iterate over the array once again and using the rows and cols sets,
        // update the elements.
        for (int i = 0; i < R; i++) {
            for (int j = 0; j < C; j++) {
                if (rows.Contains(i) || cols.Contains(j)) {
                    matrix[i][j] = 0;
                }
            }
        }
    }
}
